src/ILCompiler.Compiler/src/Compiler/SyntheticMethodDebugInfoProvider.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Internal.IL;
+using Internal.TypeSystem;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Debug information provider that generates IL assembly listing for compiler-generated
+    /// IL stubs.
+    /// </summary>
+    public class SyntheticMethodDebugInfoProvider : DebugInformationProvider, IDisposable
```

Comment:
This is not a good name for this class. Maybe something like MethodDebugInfoProviderForAllMethodsIncludingOnesWithOnlyDynamicallyGeneratedILBodies... Ok, on second thought, that name isn't good either, but this one implies its only doing stuff for synthetic methods, and that's simply wrong. Also, please update the comment.